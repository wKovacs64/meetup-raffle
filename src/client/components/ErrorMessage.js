import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({
  title,
  titleClasses,
  subtitle,
  subtitleClasses,
  problemText,
  problemTextClasses,
  ...props
}) => (
  <section {...props}>
    <span className={titleClasses}>{title}</span>
    <p className={subtitleClasses}>{subtitle}</p>
    <p className={problemTextClasses}>{problemText}</p>
  </section>
);

ErrorMessage.propTypes = {
  title: PropTypes.string,
  titleClasses: PropTypes.string,
  subtitle: PropTypes.string,
  subtitleClasses: PropTypes.string,
  problemText: PropTypes.string,
  problemTextClasses: PropTypes.string,
};

ErrorMessage.defaultProps = {
  title: 'Oops!',
  titleClasses: 'f4 b',
  subtitle: 'An error was encountered:',
  subtitleClasses: '',
  problemText: 'Unknown.',
  problemTextClasses: 'bg-red white courier pa2',
};

ErrorMessage.displayName = 'ErrorMessage';

export default ErrorMessage;
