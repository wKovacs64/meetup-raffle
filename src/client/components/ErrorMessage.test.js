import React from 'react';
import { render } from 'react-testing-library';
import ErrorMessage from './ErrorMessage';

const errorProps = {
  title: 'Test error title',
  subtitle: 'Test error subtitle',
  problemText: 'Test error problem text',
};

describe('ErrorMessage', () => {
  const { container, getByText } = render(<ErrorMessage />);

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders defaults', () => {
    render(<ErrorMessage />, { container });
    expect(getByText(ErrorMessage.defaultProps.title)).toBeTruthy();
    expect(getByText(ErrorMessage.defaultProps.subtitle)).toBeTruthy();
    expect(getByText(ErrorMessage.defaultProps.problemText)).toBeTruthy();
  });

  it('renders overrides', () => {
    render(<ErrorMessage {...errorProps} />, { container });
    expect(getByText(errorProps.title)).toBeTruthy();
    expect(getByText(errorProps.subtitle)).toBeTruthy();
    expect(getByText(errorProps.problemText)).toBeTruthy();
  });
});
