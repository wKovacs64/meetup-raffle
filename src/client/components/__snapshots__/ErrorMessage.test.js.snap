// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders 1`] = `
<div>
  <section>
    <span
      class="f4 b"
    >
      Oops!
    </span>
    <p
      class=""
    >
      An error was encountered:
    </p>
    <p
      class="bg-red white courier pa2"
    >
      Unknown.
    </p>
  </section>
</div>
`;
